flink-kafka-example/README.md

```diff
@@ -0,0 +1,75 @@
+# Flink & Kafka End-to-End Test Example
+
+## Purpose
+
+This example demonstrates how to create and run an end-to-end integration 
+test with Kafka and Flink using Kurtosis and the Kurtosis Starlark API, 
+covering these common end-to-end testing related tasks:
+
+* starting services and waiting for them to become ready for interaction
+* testing and validating (asserting) code (features) by reading and processing json from a Kafka topic
+* evaluating data from, and interacting with, APIs running inside an enclave
+* package composition using a Starlark package inside another package
+* using a docker helper image to execute a job
+
+## Running the Example
+
+To run the example, first ensure you have the latest Kurtosis CLI installed (by completing [the installation guide](https://docs.kurtosis.com/quickstart#setup)) and then running:
+
+```shell
+kurtosis run kurtosis-package/.
+```
+
+this will create an enclave with a generated name and the services. Alternately, 
+you can assign the enclave a specific name by running:
+
+```shell
+kurtosis run kurtosis-package/. --enclave flink-kafka
+```
+
+Note, if you already have an enclave running with the name `flink-kafka` you first need to remove it 
+or provide the enclave with an ephemeral name by removing the `--enclave` switch.
+
+```shell
+kurtosis enclave rm -f flink-kafka
+```
+
+## Overview
+
+The high level steps of the test are as follows:
+
+* A Kafka and two node Flink cluster are created and the test waits for them to become ready for use 
+* A demo Flink job is uploaded and executed on the Flink cluster 
+(counts the number of times it sees a word on a topic and emits the count to another topic) 
+* Data flows into a topic (`words`) that the Flink job reads from
+* The Flink job processes the data and outputs the result on to another topic (`words-counted`).
+* The Starlark script asserts if the output on the topic is as expected, thereby validating that the job works and 
+integrates correctly with the infrastructure
+
+This example demonstrates an end-to-end testing scenario that runs in an isolated testing environment that is completely ephemeral in nature, 
```

Comment:
there is an unexpected line break after `in nature,`

Reply:
I did this to not make the line too long to read in the editor. The margin breaks the line there.